From 292edad562c80260ffb8dbf7dc57fd7280b35654 Mon Sep 17 00:00:00 2001
From: Leed Aguilar <leed.aguilar@ti.com>
Date: Tue, 29 Mar 2011 13:20:34 -0500
Subject: Revert "syslink: ipu_pm: relax syslink fixed constraints"

AV playback robustness test regression detected.

The theory was that applying the tighter constraints conditionally
could save some power in cases where the tighter constraints were
not required. Reverting this change SysLink PM will applythe tighter
constraints when Ducati SS is active rather than relying on others
to do it conditionally

Change-Id: Iee512db9d64d678f5fc9831845994f64b62cb31f
Reported-by: Paul Hunt <hunt@ti.com>
Signed-off-by: Leed Aguilar <leed.aguilar@ti.com>
Cc: Miguel Vadillo <vadillo@ti.com>
---
 drivers/dsp/syslink/ipu_pm/ipu_pm.c |   41 +++++++++++++---------------------
 drivers/dsp/syslink/ipu_pm/ipu_pm.h |    5 +--
 2 files changed, 18 insertions(+), 28 deletions(-)

diff --git a/drivers/dsp/syslink/ipu_pm/ipu_pm.c b/drivers/dsp/syslink/ipu_pm/ipu_pm.c
index c69c727..1ff487d 100644
--- a/drivers/dsp/syslink/ipu_pm/ipu_pm.c
+++ b/drivers/dsp/syslink/ipu_pm/ipu_pm.c
@@ -1870,7 +1870,7 @@ static inline int ipu_pm_rel_iva_hd(struct ipu_pm_object *handle,
 	if (params->pm_iva_hd_counter == 0 && params->pm_iss_counter == 0) {
 		pr_debug("Release MPU wakeup latency\n");
 		retval = omap_pm_set_max_mpu_wakeup_lat(&pm_qos_handle,
-						IPU_PM_NO_LAT_CONSTRAINT);
+						IPU_PM_NO_MPU_LAT_CONSTRAINT);
 		if (retval) {
 			pr_err("%s %d Error setting MPU cstr\n"
 							, __func__, __LINE__);
@@ -1985,7 +1985,7 @@ static inline int ipu_pm_rel_iss(struct ipu_pm_object *handle,
 	if (params->pm_iva_hd_counter == 0 && params->pm_iss_counter == 0) {
 		pr_debug("Release MPU wakeup latency\n");
 		retval = omap_pm_set_max_mpu_wakeup_lat(&pm_qos_handle,
-						IPU_PM_NO_LAT_CONSTRAINT);
+						IPU_PM_NO_MPU_LAT_CONSTRAINT);
 		if (retval) {
 			pr_err("%s %d Error setting MPU cstr\n"
 							, __func__, __LINE__);
@@ -2481,6 +2481,13 @@ int ipu_pm_save_ctx(int proc_id)
 		return 0;
 	}
 
+#ifdef CONFIG_OMAP_PM
+	retval = omap_pm_set_max_sdma_lat(&pm_qos_handle_2,
+						IPU_PM_NO_MPU_LAT_CONSTRAINT);
+	if (retval)
+		pr_info("Unable to remove cstr on IPU\n");
+#endif
+
 	/* Because of the current scheme, we need to check
 	 * if APPM3 is enable and we need to shut it down too
 	 * Sysm3 is the only want sending the hibernate message
@@ -2530,13 +2537,6 @@ int ipu_pm_save_ctx(int proc_id)
 			goto error;
 		handle->rcb_table->state_flag |= SYS_PROC_DOWN;
 
-#ifdef CONFIG_OMAP_PM
-		retval = omap_pm_set_max_sdma_lat(&pm_qos_handle_2,
-						IPU_PM_NO_LAT_CONSTRAINT);
-		if (retval)
-			pr_info("Unable to remove cstr on IPU\n");
-#endif
-
 		/* If there is a message in the mbox restore
 		 * immediately after save.
 		 */
@@ -2613,6 +2613,12 @@ int ipu_pm_restore_ctx(int proc_id)
 		cm_write_mod_reg(HW_AUTO,
 				 OMAP4430_CM2_CORE_MOD,
 				 OMAP4_CM_DUCATI_CLKSTCTRL_OFFSET);
+#ifdef CONFIG_OMAP_PM
+		retval = omap_pm_set_max_sdma_lat(&pm_qos_handle_2,
+						IPU_PM_MM_MPU_LAT_CONSTRAINT);
+		if (retval)
+			pr_info("Unable to set cstr on IPU\n");
+#endif
 	}
 
 	/* Check if the M3 was loaded */
@@ -2666,7 +2672,7 @@ int ipu_pm_restore_ctx(int proc_id)
 #endif
 #ifdef CONFIG_OMAP_PM
 		retval = omap_pm_set_max_sdma_lat(&pm_qos_handle_2,
-						IPU_PM_MM_SDMA_LAT_CONSTRAINT);
+						IPU_PM_MM_MPU_LAT_CONSTRAINT);
 		if (retval)
 			pr_info("Unable to set cstr on IPU\n");
 #endif
@@ -2794,13 +2800,6 @@ int ipu_pm_setup(struct ipu_pm_config *cfg)
 #endif
 	BLOCKING_INIT_NOTIFIER_HEAD(&ipu_pm_notifier);
 
-#ifdef CONFIG_OMAP_PM
-	retval = omap_pm_set_max_sdma_lat(&pm_qos_handle_2,
-						IPU_PM_MM_SDMA_LAT_CONSTRAINT);
-	if (retval)
-		pr_info("Unable to set cstr on IPU\n");
-#endif
-
 	return retval;
 exit:
 	pr_err("ipu_pm_setup failed! retval = 0x%x", retval);
@@ -3087,14 +3086,6 @@ int ipu_pm_destroy(void)
 exit:
 	if (retval < 0)
 		pr_err("ipu_pm_destroy failed, retval: %x\n", retval);
-
-#ifdef CONFIG_OMAP_PM
-	retval = omap_pm_set_max_sdma_lat(&pm_qos_handle_2,
-						IPU_PM_NO_LAT_CONSTRAINT);
-	if (retval)
-		pr_info("Unable to remove cstr on IPU\n");
-#endif
-
 	return retval;
 }
 EXPORT_SYMBOL(ipu_pm_destroy);
diff --git a/drivers/dsp/syslink/ipu_pm/ipu_pm.h b/drivers/dsp/syslink/ipu_pm/ipu_pm.h
index 6093023..4ed30f4 100644
--- a/drivers/dsp/syslink/ipu_pm/ipu_pm.h
+++ b/drivers/dsp/syslink/ipu_pm/ipu_pm.h
@@ -110,9 +110,8 @@
 #define PM_CSTR_LAT_MASK	0x00000002
 #define PM_CSTR_BW_MASK		0x00000004
 
-#define IPU_PM_MM_MPU_LAT_CONSTRAINT	400
-#define IPU_PM_MM_SDMA_LAT_CONSTRAINT	400
-#define IPU_PM_NO_LAT_CONSTRAINT	-1
+#define IPU_PM_MM_MPU_LAT_CONSTRAINT	10
+#define IPU_PM_NO_MPU_LAT_CONSTRAINT	-1
 #define NO_FREQ_CONSTRAINT		0
 #define NO_LAT_CONSTRAINT		-1
 #define NO_BW_CONSTRAINT		-1
-- 
1.6.0.4

